akka-distributed-data/src/main/scala/akka/cluster/ddata/ORMap.scala

```diff
@@ -150,7 +270,16 @@ final class ORMap[A, B <: ReplicatedData] private[akka] (
    * INTERNAL API
    */
   @InternalApi private[akka] def remove(node: UniqueAddress, key: A): ORMap[A, B] = {
-    new ORMap(keys.remove(node, key), values - key)
+    // for removals the delta values map should be empty
+    val newDelta = delta match {
+      case Some(d) ⇒
+        d.merge(RemoveDeltaOp(keys.resetDelta.remove(node, key).delta.get, Map.empty[A, B], node))
+      case None ⇒
+        RemoveDeltaOp(keys.resetDelta.remove(node, key).delta.get, Map.empty[A, B], node)
+    }
+    // tombstone value for the key being removed
+    val newTombstoneVals = if (values.contains(key)) tombstoneVals + (key → values(key)) else tombstoneVals
```

Comment:
We need to keep tombstone of removed value in order to merge it later with anything that comes in an update from the node that does not know about removal.